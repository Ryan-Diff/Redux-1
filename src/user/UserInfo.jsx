import React, { Profiler } from 'react';
import { useSelector } from 'react-redux';

const UserInfo = () => {
    const userProfile = useSelector(state => state.userProfile);
    const userRepos = useSelector(state => state.userRepos);
    const loading = useSelector(state => state.loading);

    if (loading) return <h1>Loading...</h1>;
    const repoElements = userRepos.map(userRepo => (
        <li key={userRepos.name}>
            <p>{userRepo.name}</p>
            <a href={userRepo.url}>Link: {userRepo.url}:</a>
        </li>
    ));

    return (
        <>
            <h1>{userProfile.name}</h1>
            <a href={userProfile.userProfile}>GitHub Link: {userProfile.userProfile}</a>
            <p>Followers: {userProfile.followerCount}</p>
            <p>Following: {userProfile.followingCount}</p>
            <ul>
                Repositories: 
                {repoElements}
            </ul>
        </>
    );
};

export default UserInfo;