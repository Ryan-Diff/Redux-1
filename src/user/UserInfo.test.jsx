import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserInfo from './UserInfo';
import { Provider } from 'react-redux';
import store from '../store';

describe('UserInfo component', () => {
    afterEach(() => cleanup());

    it('Renders a users GitHub info', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <UserInfo />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
