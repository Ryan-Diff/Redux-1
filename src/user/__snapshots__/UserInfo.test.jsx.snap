// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component Renders a users GitHub info 1`] = `
<DocumentFragment>
  <h1 />
  <a>
    GitHub Link: 
  </a>
  <p>
    Followers: 
  </p>
  <p>
    Following: 
  </p>
  <ul>
    Repositories:
  </ul>
</DocumentFragment>
`;
